### PR TITLE
Add UseRbacAuth and EnableInteractiveCredentials to Azure Blob Storage Extension

### PR DESCRIPTION
**Pull request by philnach**
Implement UseRbacAuth Azure Blob Storage Extension enhancement proposed in #135

### CODE REVIEW

File: Extensions/AzureBlob/Cosmos.DataTransfer.AzureBlobStorage/AzureBlobSourceSettings.cs [hunk 1]
```diff
@@ -17,4 +18,21 @@ public class AzureBlobSourceSettings : IDataExtensionSettings
     public string BlobName { get; set; } = null!;
 
     public int? ReadBufferSizeInKB { get; set; }
+
+    public bool UseRbacAuth { get; set; }
+
+    public bool EnableInteractiveCredentials { get; set; } = true;
```
Comment (markjbrown) [inline Extensions/AzureBlob/Cosmos.DataTransfer.AzureBlobStorage/AzureBlobSourceSettings.cs:24]: I'm not sure this makes sense for this to default as true

File: Extensions/AzureBlob/README.md [hunk 2]
```diff
@@ -6,7 +6,10 @@ The Azure Blob Storage extension provides reading and writing of formatted files
 
 ## Settings
 
-Source and Sink settings require the parameters shown below. 
+Source and sink require settings used to locate and access the Azure Blob Storage account. This can be done in one of two ways:
+
+- Using a `ConnectionString` that includes an AccountEndpoint and AccountKey
+- Using RBAC (Role Based Access Control) by setting `UseRbacAuth` to true and specifying `AccountEndpoint` and optionally `EnableInteractiveCredentials` to prompt the user to log in to Azure if default credentials are not available.
```
Comment (markjbrown) [inline Extensions/AzureBlob/README.md:12]: See my earlier comment on this defaulting EnableInteractiveCredentials to true. Not having it default as true is consistent with the description here.

File: Extensions/Cosmos/README.md [hunk 3]
```diff
@@ -7,10 +7,12 @@ The Cosmos data transfer extension provides source and sink capabilities for rea
 ## Settings
 
 Source and sink require settings used to locate and access the Cosmos DB account. This can be done in one of two ways:
+
 - Using a `ConnectionString` that includes an AccountEndpoint and AccountKey
-- Using RBAC (Role Based Access Control) by setting `UseRbacAuth` to true and specifying `AccountEndpoint` and optionally `EnableInteractiveCredentials` to prompt the user to log in to Azure if default credentials are not available.
+- Using RBAC (Role Based Access Control) by setting `UseRbacAuth` to true and specifying `AccountEndpoint` and optionally `EnableInteractiveCredentials` to prompt the user to log in to Azure if default credentials are not available. See ([migrate-passwordless](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/migrate-passwordless?tabs=sign-in-azure-cli%2Cdotnet%2Cazure-portal-create%2Cazure-portal-associate%2Capp-service-identity) for how to configure Cosmos DB for passwordless access.
```
Comment (markjbrown) [inline Extensions/Cosmos/README.md:12]: Can you remove the language-locale in this URL? If there is a translated version of this doc MS Learn will direct the user to it vs. sending to US English only.

**Comment by markjbrown** [comment]
Thank you @philnach for this PR. Overall this looks good. Left some comments to resolve. Thanks.

File: Extensions/AzureBlob/Cosmos.DataTransfer.AzureBlobStorage/AzureBlobSourceSettings.cs (hunk 1 above)
Comment (philnach) [reply Extensions/AzureBlob/Cosmos.DataTransfer.AzureBlobStorage/AzureBlobSourceSettings.cs:24]: Thanks for the feedback, I agree. I followed the same [pattern](https://github.com/AzureCosmosDB/data-migration-desktop-tool/blob/df9f09dce93491f60f2a7356867710bf1820d5af/Extensions/Cosmos/Cosmos.DataTransfer.CosmosExtension/CosmosSettingsBase.cs#L17C1-L18C1) of defaulting to true as the Cosmos extension. Should I remove the default from the Cosmos extension as well?

File: Extensions/Cosmos/README.md (hunk 3 above)
Comment (philnach) [reply Extensions/Cosmos/README.md:12]: Good catch. I had a tough time finding any resource that indicated how to link to Microsoft Learn documentation. I did find a mention of the don't include locale in [Use Links in documentation](https://learn.microsoft.com/contribute/content/how-to-write-links#links-to-pages-that-are-not-part-of-the-technical-documentation-set). I'll remove the en-us.

File: Extensions/AzureBlob/Cosmos.DataTransfer.AzureBlobStorage/AzureBlobSourceSettings.cs (hunk 1 above)
Comment (philnach) [reply Extensions/AzureBlob/Cosmos.DataTransfer.AzureBlobStorage/AzureBlobSourceSettings.cs]: I went ahead and removed true default from both the Azure Blob Storage extension and the Cosmos extension, figuring not at least it aligns with the documentation.

File: Extensions/Cosmos/README.md (hunk 3 above)
Comment (philnach) [reply Extensions/Cosmos/README.md]: en-us removed.

File: Extensions/AzureBlob/README.md (hunk 2 above)
Comment (philnach) [reply Extensions/AzureBlob/README.md:12]: default of true has been removed. documentation is now correct.

**Review by markjbrown — DISMISSED** [review]

**Review by markjbrown — APPROVED** [review]

**Pull request merged**